### ISSUE TITLE
eslint-plugin-vue rules don't work with composition API

### ISSUE THREAD

**Issue by caryuval** [vuejs/eslint-plugin-vue#1461]
Hi all,

I really appreciate all the work you did to create this plugin, it's truly amazing!

I found an issue while using this plugin together with Vue 2 and the composition API.
It seems like it doesn't detect issues inside the script tag when using the composition API.

I get no warning for this:
```
<template>
  <div class=""></div>
</template>

<script lang="ts">
  import { createComponent } from '@vue/composition-api';
  import AInput from '@/vue/components/a_input/AInput.vue';

  export default createComponent({
    name: '',
    props: {},
    components: { AInput },
    setup() {
      return {};
    },
  });
</script>
```

I do get warnings when not using the composition API:
Warnings:
- The "components" property should be above the "props" property on line 10.eslintvue/order-in-components
- The "AInput" component has been registered but not used.eslintvue/no-unused-components
```
<template>
  <div class=""></div>
</template>

<script lang="ts">
  import AInput from '@/vue/components/at_input/AInput.vue';

  export default {
    name: '',
    components: { AInput },
    props: {},
    setup() {
      return {};
    },
  };
</script>
```

I noticed in this section https://eslint.vuejs.org/user-guide/#how-does-eslint-detect-components
That `createComponent` of the composition API is not included.
Is it possible to include this?
Does it work out of the box in Vue 3 or there likely to be issues there as well?

Dependencies:
- "eslint-plugin-vue": "^7.0.0-beta.2"
- "eslint": "^7.7.0"
- "@vue/composition-api": "^0.3.4"
- "vue": "^2.6.11"

.eslintrc.js
```
module.exports = {
  root: true,
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: '@typescript-eslint/parser',
    sourceType: 'module',
  },
  plugins: ['@typescript-eslint'],
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'prettier',
    'prettier/vue',
    'prettier/@typescript-eslint',
    'plugin:vue/base',
    'plugin:vue/essential',
    'plugin:vue/strongly-recommended',
    'plugin:vue/recommended',
    'plugin:vue/vue3-essential',
    'plugin:vue/vue3-strongly-recommended',
    'plugin:vue/vue3-recommended',
  ],
  rules: {
    'vue/component-name-in-template-casing': [
      'error',
      'PascalCase',
      {
        registeredComponentsOnly: false,
      },
    ],
    'vue/no-unused-components': [
      'error',
      {
        ignoreWhenBindingPresent: true,
      },
    ],
  },
  overrides: [
    {
      files: '*.{ts,js,vue}',
      rules: {
        'vue/max-attributes-per-line': 'off',
        'vue/html-self-closing': 'off',
        'vue/singleline-html-element-content-newline': 'off',
        'vue/html-closing-bracket-newline': 'off',
        'vue/no-deprecated-dollar-listeners-api': 'off',
        'vue/attributes-order': 'off',
      },
    },
  ],
};
```

Thanks a lot for your help!

**Comment by ota-meshi** [comment]
`createComponent` is a deprecated feature. So we don't support it. Please upgrade `@vue/composition-api` and use `defineComponent`.

https://github.com/vuejs/composition-api/blob/master/CHANGELOG.md#100-beta1-2020-06-30
